docs/IsolationNetworkPolicies.md

```diff
@@ -0,0 +1,210 @@
+# Isolation 
+
+_**Note:**_ below we start with Network Polices as an isolation method, later we will add other methods.
+
+_TODO_: add links to other options.
+
+## Introduction
+
+In the current Fybrik implementation the entry point of a data plane is published as a part of the `FybrikApplication` 
+status, it is completely unprotected, and as result everyone can connect to the data plane. If a Fybrik data plane 
+consist of a module chain, the same vulnerability exists for the chained modules.  
+In `Read` flow cases it leads to governance/security violations, `Write` and `Delete` flows can create severe accidents, 
+such as unauthorized data changes or deletions. Given that we have to provide a mechanism, which will prevent 
+unauthorized data plane access.
+
+## Challenge
+
+When we talk about Fybrik data plane isolation, we are talking about connectivity between a user workload, which is out of
+Fybrik scope, and Fybrik modules, which can be developed by third parties. Therefore, in this document (and in its
+extensions) we provide our recommendations how to protect data plane access and possible solutions. Responsibility of the
+data plane isolation is shared between Fybrik core, Fybrik users, module developers and probably IT teams.
+
+- Fybrik users should provide all relevant information about possible workload locations to Fybrik, correctly 
+configure their workloads. For example, to set labels on the workload and inform Fybrik about the labels and namespace 
+where the workload runs.
+
+- Depends on the isolation method, Fybrik should propagate all relevant information to the deployed modules, and/or
+correctly configure a chosen protection mechanism, e.g. Network Policies..
+
+- Module developers should allow to protect the module instances. For example, to allow TLS support 
+ 
+- In some complicated cases, IT teams interfere can be required, e.g. to configure external access points, firewalls and so on.
+
+## Requirements
+
+- Only predefined users/workloads should be able to access the data plane.
+- If a data plane consist of a module chain we should prevent authorized access to the intermediate modules too.
+- Modules should not be able to "illegally" communicate with modules outside of the defined data plane(s) and should 
+connect to only predefined data sets.
```

Comment:
I don't see how network policies prevent the module from connecting a different dataset. Is perhaps giving them credentials to specific datasets only enough?

Reply:
We can prevent connectivity to a different destination. (Egress restriction)